src/inmanta/server/services/resourceservice.py

```diff
@@ -416,6 +417,108 @@ async def mark_deployed(
             connection=connection,
         )
 
+    async def _update_deploy_state(
+        self,
+        env: data.Environment,
+        resource_id: ResourceIdStr,
+        timestamp: datetime.datetime,
+        version: int,
+        status: ResourceState,
+        message: str,
+        fail_on_error: bool,
+        is_increment_notification: bool,
+        connection: Optional[Connection] = None,
+    ) -> None:
+        """
+        Factored out the code to set a status on a resource
```

Comment:
I don't feel like this is very relevant, especially as the first sentence. I think it would be better to focus on what the method does, and then perhaps end with that the implementation is factored out of / a subset of what `resource_action_update` does and where it differs (or which specific parts of it it addresses), and optionally the motivation for making it a separate method.